### ISSUE TITLE
Unexpected behaviour with JAX version

### ISSUE THREAD

**Issue by itk22** [patrick-kidger/optimistix#60]
Hi Patrick, 
I have run into an peculiar behaviour when trying to install optimistix with jax 0.4.25: 

![image](https://github.com/patrick-kidger/optimistix/assets/112338249/e7150693-3e71-478a-b1cf-02371610686a)

It seems to upgrade to 0.4.28 despite 0.4.25 already satisfying the requirement. Is this the expected behaviour?

**Comment by itk22** [comment]
I think I might have found the reason - the newest version of lineax require jax 0.4.26 which is quite new. Is there any particular reason for this?
![image](https://github.com/patrick-kidger/optimistix/assets/112338249/415f6132-62ec-4589-b670-95968161cf5a)

**Comment by patrick-kidger** [comment]
Yup. JAX version 0.4.26 fixed a correctness issue when using complex numbers, that we were hitting in Lineax.

If you really need to use an earlier version, and you're not using complex numbers, then you can probably get away with it though :)

**Comment by itk22** [comment]
Ahh, I see. Thanks for the super fast response!
